### PR TITLE
Add attenuated MIP (attenuated_mip) rendering mode to `VolumeVisual`

### PR DESCRIPTION
**Pull request by alisterburt**
This PR upstreams the attenuated maximum intensity projection rendering mode from napari/napari#846 (by @sofroniewn) into vispy

### CODE REVIEW

File: vispy/visuals/volume.py [hunk 1]
```diff
@@ -749,6 +779,17 @@ def threshold(self, value):
...
 
+    @property
+    def attenuation(self):
+        """The attenuation rate to apply for the attenuated mip render method."""
+        return self._attenuation
+
+    @attenuation.setter
+    def attenuation(self, value):
+        self._attenuation = float(value)
+        self.shared_program['u_attenuation'] = self._attenuation
```
Comment (djhoese) [inline vispy/visuals/volume.py:790]: This should probably have the `if 'u_attenuation' ...` like threshold's setter has.

File: vispy/visuals/volume.py [hunk 2]
```diff
@@ -728,6 +756,8 @@ def method(self, method):
         # Get rid of specific variables - they may become invalid
         if 'u_threshold' in self.shared_program:
             self.shared_program['u_threshold'] = None
+        elif 'u_attenuation' in self.shared_program:
```
Comment (djhoese) [inline vispy/visuals/volume.py:759]: Can this maybe be made an `if` instead of `elif` for future flexibility? Otherwise this assumes that two shaders will never use both parameters. I know, nit-pick, but still.

**Review by djhoese — CHANGES_REQUESTED** [review]
Just some small changes, but otherwise looks pretty good.

**Comment by sofroniewn** [comment]
This is looking great @alisterburt! It will let us remove our vendored copy of the volume visual from napari

File: vispy/visuals/volume.py [hunk 3]
```diff
@@ -707,8 +737,6 @@ def method(self):
               the result is opaque.
             * mip: maxiumum intensity projection. Cast a ray and display the
               maximum value that was encountered.
-            * minip: minimum intensity projection. Cast a ray and display the
-              minimum value that was encountered.
```
Comment (almarklein) [inline vispy/visuals/volume.py:711]: Why are these lines removed?

**Review by almarklein — COMMENTED** [review]
Nice!

File: vispy/visuals/volume.py (hunk 3 above)
Comment (alisterburt) [reply vispy/visuals/volume.py:711]: whoops - merge shenanigans, will fix!

**Comment by alisterburt** [comment]
thanks for the reviews @djhoese, @almarklein and @sofroniewn - much appreciated

I've updated to incorporate suggested changes, rebased onto master and fixed docstrings, assuming all tests pass I think we're good to go!

File: vispy/visuals/volume.py [hunk 4]
```diff
@@ -731,6 +762,11 @@ def method(self):
               maximum value that was encountered.
             * minip: minimum intensity projection. Cast a ray and display the
               minimum value that was encountered.
+            * attenuated_mip: attenuated maximum intensity projection. Cast a
+              ray and display the maximum value encountered. Values are
+              attenuated as the ray moves deeper into the volume.
+            * minip: minimum intensity projection. Cast a ray and display the
+              minimum value that was encountered.
             * additive: voxel colors are added along the view ray until
```
Comment (djhoese) [inline vispy/visuals/volume.py:770]:
```suggestion
            * attenuated_mip: attenuated maximum intensity projection. Cast a
              ray and display the maximum value encountered. Values are
              attenuated as the ray moves deeper into the volume.
            * additive: voxel colors are added along the view ray until
```

**Review by djhoese — APPROVED** [review]
I fixed one small thing (minip was listed twice in the docstring). Looks good now. It bothers me a little that `_attenuation` isn't directly declared in the `__init__` but threshold works the same way so :shrug: we'll ignore it for now. Thanks.

**Comment by alisterburt** [comment]
thanks @djhoese for handling this so smoothly!
